nibabel/streamlines/trk.py

```diff
@@ -423,8 +424,20 @@ def save(self, fileobj):
             i4_dtype = np.dtype("<i4")  # Always save in little-endian.
             f4_dtype = np.dtype("<f4")  # Always save in little-endian.
 
+            # Make sure streamlines are in rasmm then send them to voxmm.
```

Comment:
Could you say a bit more about this change?

Reply:
Because `self.tractogram` could be a LazyTractogram, peeking the first element requires to call `iter()` on it (i.e. start looping through the TractogramItems), but later I have to send the streamlines to voxmm, that is calling `apply_affine` on `self.tractogram` which would have been advanced by one element (because of peek_next).    

The alternative would have been to create a new LazyTractogram from the iterable return by peek_next, then call the apply_affine function. I just thought it would be simpler to do the transformation first. It shouldn't change the overall logic of the method: peek element to get some data type information, apply affine and save to disk.

Reply:
Could you add something like this as an explanation in the code?

Reply:
Better?